src/ccl_power.c

```diff
@@ -13,6 +13,43 @@
 #include "ccl_emu17.h"
 #include "ccl_emu17_params.h"
 
+/* BCM correction */
+// See Schneider & Teyssier (2015) for details of the model.
+double ccl_bcm_model_fka(ccl_cosmology * cosmo, double k, double a, int *status){
+
+  double fkz;
+  double b0;
+  double bfunc, bfunc4;
+  double kg;
+  double gf,scomp;
+  double kh;
+  double z;
+
+  z=1./a-1.;
+  kh = k/cosmo->params.h;
+  b0 = 0.105*cosmo->params.bcm_log10Mc-1.27;
+  bfunc = b0/(1.+pow(z/2.3,2.5));
+  bfunc4 = (1-bfunc)*(1-bfunc)*(1-bfunc)*(1-bfunc);
+  kg = 0.7*bfunc4*pow(cosmo->params.bcm_etab,-1.6);
+  gf = bfunc/(1+pow(kh/kg,3.))+1.-bfunc; //k in h/Mpc
+  scomp = 1+(kh/cosmo->params.bcm_ks)*(kh/cosmo->params.bcm_ks); //k in h/Mpc
+  fkz = gf*scomp;
+  return fkz;
+}
+
+static void correct_bcm(ccl_cosmology *cosmo,
+			int na,double *a_arr,int nk,double *lk_arr,double *pk2d,
+			int *status)
+{
+  for(int ii=0;ii<na;ii++) {
+    double a=a_arr[ii];
+    for(int jj=0;jj<nk;jj++) {
+      double k=exp(lk_arr[jj]);
+      double fbcm=ccl_bcm_model_fka(cosmo,k,a,status);
+      pk2d[jj+nk*ii]+=log(fbcm);
```

Comment:
So the `correct_bcm` method applies corrections to its pk2d input.

Reply:
correct, but we only pass pk2d holding non-linear power spectra to it.